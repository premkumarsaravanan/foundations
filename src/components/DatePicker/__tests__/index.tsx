import * as React from 'react'
import { shallow, mount } from 'enzyme'
import { DatePicker, DatePickerProps } from '../index'
import { Formik, Form } from 'formik'
import toJson from 'enzyme-to-json'

const props: DatePickerProps = {
  name: 'abc',
  labelText: 'abc',
  id: '123'
}

describe('Checkbox', () => {
  it('should match a snapshot', () => {
    expect(toJson(shallow(<DatePicker {...props} />))).toMatchSnapshot()
  })

  it('should work when integrating with Formik', () => {
    const wrapper = mount(
      <Formik
        initialValues={{ test: false }}
        onSubmit={jest.fn()}
        render={() => (
          <Form>
            <DatePicker {...props} />
          </Form>
        )}
      />
    )
    expect(wrapper.find('label')).toHaveLength(1)
  })

  afterEach(() => {
    jest.resetAllMocks()
  })
})
