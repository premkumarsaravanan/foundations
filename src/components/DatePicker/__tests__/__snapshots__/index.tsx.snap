// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should match a snapshot 1`] = `
<FormikConnect(FieldInner)
  name="abc"
  render={[Function]}
/>
`;
